SingularityUI/app/components/globalSearch/GlobalSearch.cjsx

```diff
@@ -0,0 +1,52 @@
+React = require 'react'
+
+{ Typeahead } = require 'react-typeahead'
+
+
+GlobalSearch = React.createClass
+  optionSelected: (requestId) ->
+    app.router.navigate "/request/#{ requestId }", { trigger: true }
+    @clear()
+    @props.onHide()
+
+  clear: ->
+    @refs.typeahead.setEntryText('')
+    @refs.typeahead._onEscape() # hack to clear search index, TODO: PR react-typeahead
+
+  focus: ->
+    @refs.typeahead.focus()
+
+  render: ->
+    if @props.visible
+      @focus()
+
+    options = _.pluck(@props.requests.toJSON(), 'id')
+
+    globalSearchClasses = [
+      'global-search'
+      'global-search-active' if @props.visible
+    ].join(' ')
```

Comment:
easier to use the `classnames` library for this (example: https://github.com/HubSpot/Singularity/blob/master/SingularityUI/app/components/logs/LoadingSpinner.cjsx#L9-L12)